resources/lib/vrtplayer/vrtapihelper.py

```diff
@@ -166,13 +166,17 @@ def _map_to_season_items(self, api_url, buckets, episode):
         metadata_creator = metadatacreator.MetadataCreator()
         metadata_creator.mediatype = 'season'
         for bucket in buckets:
-            season_title = bucket.get('key')
-            title = ''.join((self._kodi_wrapper.get_localized_string(32094), ' ', season_title))
+            season_key = bucket.get('key')
+            season_size = str(bucket.get('doc_count'))
+            title = ''.join((self._kodi_wrapper.get_localized_string(32094), ' ', season_key))
             season_facet = '&facets[seasonTitle]='
-            if ' ' in season_title:
-                season_title = season_title.replace(' ', '-')
+            # Workaround: seasonTitle facet doesn't support spaces and hyphens properly, luckily seasonName facet escapes this
```

Comment:
To me this is not clear. So I doubt anyone else reading this code will understand why this is needed.
Maybe it would help if we add examples in the comments for each case.

Also, why don't we *always* replace ' ' with '-', and always check for seasonKey instead ?

Reply:
So my implementation would simply do this:
```python
            season_title = bucket.get('key')
            season_size = str(bucket.get('doc_count'))
            title = ''.join((self._kodi_wrapper.get_localized_string(32094), ' ', season_title))

            # Workaround: seasonTitle facet doesn't support spaces and hyphens properly, luckily seasonName facet escapes this
            season_key = season_title.replace(' ', '-')
            season_facet = '&facets[seasonName]='
```

Reply:
We get the available season values from faceted search api with `bucket.get('key')`, and **these values always match the `seasonTitle` values**.
For instance: https://vrtnu-api.vrt.be/search?i=video&facets[programUrl]=//www.vrt.be/vrtnu/a-z/girls-talk/
We get value `1`, and `2`
Then we filter the episodes from season 1 with the following request: https://vrtnu-api.vrt.be/search?i=video&facets[programUrl]=//www.vrt.be/vrtnu/a-z/girls-talk/&facets[seasonTitle]=1

We get 5 episodes, this is correct.

And now, try to filter with `seasonName`: https://vrtnu-api.vrt.be/search?i=video&facets[programUrl]=//www.vrt.be/vrtnu/a-z/girls-talk/&facets[seasonName]=1

We get 0 episodes, because the value of `seasonName` is `2018` and doesn't match the `bucket.get('key')` value, wich is `1`.

So, `seasonName` isn't a reliable key to get the episodes from the corresponding season.

Reply:
Thanks for the detailed explanation.

I wonder if there's no better way to escape those characters, rather than assume the seasonName will related to the seasonTitle (when spaces or dashes are encountered).
I tried various possibilities to escape (or do exact matches) using seasonName, but it seems to be a starting match (so `facets[seasonTitle]=2` or `facets[seasonTitle]=201` return the same thing).

Reply:
I found a way to maximise the use of `seasonName`.

Reply:
While testing various possible syntaxis I got this error:
```
TypeError: facet_value.indexOf is not a function
   at module.exports (/var/app/current/lib/queryBuilders/vrt-video/search.js:83:37)
   at Object.makeESQuery (/var/app/current/lib/queryBuilders/index.js:6:20)
   at handler (/var/app/current/lib/requestHandlers/search.js:21:42)
   at Layer.handle [as handle_request] (/var/app/current/node_modules/express/lib/router/layer.js:95:5)
   at trim_prefix (/var/app/current/node_modules/express/lib/router/index.js:312:13)
   at /var/app/current/node_modules/express/lib/router/index.js:280:7
   at Function.process_params (/var/app/current/node_modules/express/lib/router/index.js:330:12)
   at next (/var/app/current/node_modules/express/lib/router/index.js:271:10)
   at cors (/var/app/current/node_modules/cors/lib/index.js:179:7)
   at /var/app/current/node_modules/cors/lib/index.js:229:17
```
trying: https://vrtnu-api.vrt.be/search?i=video&facets[programUrl]=//www.vrt.be/vrtnu/a-z/universiteit-van-vlaanderen/&facets=seasonTitle:201
This gives a little insight into what was used for the API frontend, i.e. nodejs and queryBuilders